shared/rocroller/lib/source/KernelGraph/Transformations/AddPrefetch.cpp

```diff
@@ -967,17 +997,36 @@ namespace rocRoller
                 auto exchanges
                     = graph.control.findNodes(bodies, isExchangePredicate).to<std::vector>();
 
-                for(auto exchangeTag : exchanges)
+                std::map<int, int> scaleLoadU;
+
+                for(auto const exchangeTag : exchanges)
                 {
                     auto prefetchGlobalU
                         = (m_exchangeSegment[exchangeTag] + numInFlight) % numUnroll;
 
+                    auto loadTag = getLoadForExchange(exchangeTag, graph);
+                    AssertFatal(loadTag.has_value(),
+                                "couldn't find the load associated with the exchange");
+
+                    if(auto const search = scaleLoadU.find(loadTag.value());
+                       search != scaleLoadU.end())
+                    {
+                        if(search->second > prefetchGlobalU)
+                            scaleLoadU[loadTag.value()] = prefetchGlobalU;
+                    }
+                    else
+                        scaleLoadU[loadTag.value()] = prefetchGlobalU;
```

Comment:
nit, logic simplification and formatting:
```suggestion
                    auto const search = scaleLoadU.find(loadTag.value());
                    if(search == scaleLoadU.end() or search->second > prefetchGlobalU)
                        scaleLoadU[loadTag.value()] = prefetchGlobalU;

```